pkg/handlers/v1/nexposeassetproducer.go

```diff
@@ -38,6 +38,9 @@ func (h *NexposeScannedAssetProducer) Handle(ctx context.Context, in ScanInfo) {
 			if !ok {
 				assetChan = nil
 			} else {
+				if asset == (domain.AssetEvent{}) {
```

Comment:
question: why handle this here rather than in the `AssetFetcher`?

Reply:
No reason other than we're checking for empty assets once here, but we'd have to check in both FetchAssets() and makeRequest() since both of those functions put assets on the assetChannel.

Reply:
Yeah, I agree with Kevin here. As long as empty assets are never useful - and I can't think of any situation in which they would be - part of the contract of the AssetFetcher interface (that can be called out in the godoc comments for FetchAsset()) are that each asset returned is valid. I do think the validation logic belongs there.

Reply:
Okay, I'll move it to the AssetFetcher. Thanks y'all.